java/cuvs-java/src/main/java22/com/nvidia/cuvs/internal/CuVSMatrixBaseImpl.java

```diff
@@ -15,22 +15,36 @@
  */
 package com.nvidia.cuvs.internal;
 
+import static com.nvidia.cuvs.internal.common.LinkerHelper.C_CHAR;
+import static com.nvidia.cuvs.internal.common.LinkerHelper.C_FLOAT;
+import static com.nvidia.cuvs.internal.common.LinkerHelper.C_INT;
 import static com.nvidia.cuvs.internal.panama.headers_h.*;
 
 import com.nvidia.cuvs.CuVSMatrix;
-import java.lang.foreign.Arena;
-import java.lang.foreign.MemorySegment;
+import com.nvidia.cuvs.CuVSResources;
+import com.nvidia.cuvs.internal.panama.DLDataType;
+import com.nvidia.cuvs.internal.panama.DLDevice;
+import com.nvidia.cuvs.internal.panama.DLManagedTensor;
+import com.nvidia.cuvs.internal.panama.DLTensor;
+import java.lang.foreign.*;
+import java.util.Locale;
 
 public abstract class CuVSMatrixBaseImpl implements CuVSMatrix {
   protected final MemorySegment memorySegment;
   protected final DataType dataType;
+  protected final ValueLayout valueLayout;
   protected final long size;
   protected final long columns;
 
   protected CuVSMatrixBaseImpl(
-      MemorySegment memorySegment, DataType dataType, long size, long columns) {
+      MemorySegment memorySegment,
+      DataType dataType,
+      ValueLayout valueLayout,
```

Comment:
I would have thought `ValueLayout` would be the only thing that would need to be stored here, and that the `DataType` could be queried/derived from the `ValueLayout`.

That wouldn't work for two reasons:
1. We need the datatype to distinguish between the storage of INT and UINT.
2. I can't seem to find a way to query the datatype from the `ValueLayout` anyway. 🤷 

Looks like we would need both.